web-app/src/test/com.isa.aem/calculatorMethod/CurrencyExchangeRateRepositoryTest.java

```
@@ -0,0 +1,212 @@
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.NoSuchElementException;
+
+public class CurrencyExchangeRateRepositoryTest {
+    private CurrencyRepository currencyRepository = new CurrencyRepository();
+    private FileContentReader fileContentReader = new FileContentReader();
+    private CurrencyExchangeRateRepository currencyExchangeRateRepository;
+
+    @Before
+    public void init() {
+        fileContentReader.readFile();
+        fileContentReader.addPLNToListCurrency();
+        LoadCurrencyNameCountryFlags loadCurrencyNameCountryFlags = new LoadCurrencyNameCountryFlags();
+    }
+
+    @Test
+    @DisplayName("Should return date before or equal Today's date")
+
+    public void returnsMaxDateForCorrectCurrencyName() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "PLN";
+        // act
+        LocalDate result = currencyExchangeRateRepository.getMaxDateForSelectedCurrency(currency);
+        // assert
+        assertThat(result)
+                .isBeforeOrEqualTo(LocalDate.now());
+    }
+
+    @Test
+    @DisplayName("Should return date before or equal Today's date when currency is in File even though it is write with a lowercase letter")
+
+    public void returnsMaxDateForCorrectCurrencyWhenIsWriteInLowerCase() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "pln";
+        // act
+        LocalDate result = currencyExchangeRateRepository.getMaxDateForSelectedCurrency(currency);
+        // assert
+        assertThat(result)
+                .isBeforeOrEqualTo(LocalDate.now());
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return message \"No value present\" if currency is not on the list")
+
+    public void returnsThrowWhenCurrencyNotInTheFileForMethodGetMaxDateForSelectedCurrency() {
```

nazwa tej metody jest niespojna z DisplayName i nieczytelna. to co jest w DisplayName jest bardziej czytelna ale to chyba nie jest to czego oczekujesz. z kodu wyglada na to ze oczekujesz exception a nie messsage "No value present"

---

ok. zmien tylko to co sie tyczy 'return throw'
napisalas, powinien "zwrocic rzucic kiedy cos tam"...sprobuj jedno z ponizszych: 

should throw an exception jakis tam kiedy cos tam
should return an exception jakis tam kiedy cos tam (czyli zwrocic wyjatek)

---

poprawione